src/deploy/firestore/deploy.ts

```diff
@@ -1,10 +1,42 @@
 import * as clc from "colorette";
 
 import { FirestoreApi } from "../../firestore/api";
+import * as types from "../../firestore/api-types";
 import { logger } from "../../logger";
 import * as utils from "../../utils";
 import { RulesDeploy, RulesetServiceType } from "../../rulesDeploy";
 import { IndexContext } from "./prepare";
+import { FirestoreConfig } from "../../firebaseConfig";
+import { sleep } from "../../utils";
+import { Options } from "../../options";
+
+async function createDatabase(context: any, options: Options): Promise<void> {
```

Comment:
All the nonnull assertions in here are giving code smells that we might hit accidental NPEs from some edge case here. Can we clean up the typing at all to defend against this?

Reply:
Sure, I can do that here~

The typesafe-ness in `firebase deploy` in general is quite bad. Most args are passed as `any`  right now.